clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/caching/view/model/KubernetesRawResource.java

```diff
@@ -0,0 +1,90 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.netflix.spinnaker.clouddriver.kubernetes.caching.view.model;
+
+import com.google.common.collect.ImmutableMap;
+import com.netflix.spinnaker.cats.cache.CacheData;
+import com.netflix.spinnaker.clouddriver.kubernetes.KubernetesCloudProvider;
+import com.netflix.spinnaker.clouddriver.kubernetes.caching.Keys;
+import com.netflix.spinnaker.clouddriver.kubernetes.caching.agent.KubernetesCacheDataConverter;
+import com.netflix.spinnaker.clouddriver.kubernetes.description.manifest.KubernetesApiVersion;
+import com.netflix.spinnaker.clouddriver.kubernetes.description.manifest.KubernetesKind;
+import com.netflix.spinnaker.clouddriver.kubernetes.description.manifest.KubernetesManifest;
+import com.netflix.spinnaker.clouddriver.model.*;
+import com.netflix.spinnaker.moniker.Moniker;
+import java.util.Map;
+import javax.annotation.Nullable;
+import javax.annotation.ParametersAreNonnullByDefault;
+import lombok.Value;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Value
+public final class KubernetesRawResource
+    implements KubernetesResource, RawResource, RawResourceProvider.Details {
+  private static final Logger log = LoggerFactory.getLogger(KubernetesRawResource.class);
+  private final String account;
+  private final String name;
+  private final String namespace;
+  private final String displayName;
+  private final KubernetesApiVersion apiVersion;
+  private final KubernetesKind kind;
+  private final Map<String, String> labels;
+  private final Moniker moniker;
+  private final Long createdTime;
+  private final Map<String, Object> data;
+
+  private KubernetesRawResource(KubernetesManifest manifest, String key, Moniker moniker) {
+    this.account = ((Keys.InfrastructureCacheKey) Keys.parseKey(key).get()).getAccount();
+    this.name = manifest.getFullResourceName();
+    this.displayName = manifest.getName();
+    this.apiVersion = manifest.getApiVersion();
+    this.kind = manifest.getKind();
+    this.namespace = manifest.getNamespace();
+    this.labels = ImmutableMap.copyOf(manifest.getLabels());
+    this.moniker = moniker;
+    this.createdTime = manifest.getCreationTimestampEpochMillis();
+    this.data = manifest;
+  }
+
+  @Nullable
+  @ParametersAreNonnullByDefault
+  public static KubernetesRawResource fromCacheData(CacheData cd) {
+    KubernetesManifest manifest = KubernetesCacheDataConverter.getManifest(cd);
+    if (manifest == null) {
+      log.warn("Cache data {} inserted without a manifest", cd.getId());
+      return null;
+    }
+    Moniker moniker = KubernetesCacheDataConverter.getMoniker(cd);
+    return new KubernetesRawResource(manifest, cd.getId(), moniker);
```

Comment:
Should we filter resources like:

- Deployments
- StatefulSets
- DaemonSets
- Services
- IngressControllers

Basically Resources that already have a designated tab on Spinnaker?

Reply:
I think it should be possible. We could expose the Spinnaker Kind (load balancer, firewalls, server groups) in an attribute. That way, it'd be possible to filter them out on the UI side

Reply:
Another thought would be "sensitive" resource handling, aka secrets.  Might be careful on what's exposed/how/etc.

Reply:
From what I saw, secret values are not stored in the cache because they are not returned by the regular `kubectl get -o json` call. Don't know if that depends on the Kubernetes version